packages/babel-plugin-transform-class-properties/test/fixtures/general/foobar/expected.js

```diff
@@ -8,10 +8,13 @@ var Child = function (_Parent) {
 
         var _this = babelHelpers.possibleConstructorReturn(this, (Child.__proto__ || Object.getPrototypeOf(Child)).call(this));
 
-        _this.scopedFunctionWithThis = function () {
-            _this.name = {};
-        };
-
+        Object.defineProperty(_this, "scopedFunctionWithThis", {
+            enumerable: true,
+            writable: true,
+            value: function value() {
```

Comment:
Hmm, yeah seems like we'd ideally want to implement name inference here better, though it doesn't appear to be officially in the class-properties spec text anyway: https://github.com/tc39/proposal-class-public-fields/issues/48